### PR TITLE
feat: errors v2

### PR DESCRIPTION
**Pull request by mertmit**
## Change Summary

- More specific error messages for common situations

Note: this is still in progress currently covered most used parts, eventually we will switch all places

## Change type

- [x] feat: (new feature for the user, not a new feature for build script)

## Summary by CodeRabbit

- **Refactor**
	- Improved clarity and consistency of error messages across the platform.
	- Enhanced error handling logic for various scenarios including missing records, columns, bases, views, and tables.
- **Bug Fixes**
	- Fixed issues related to incorrect error messages being displayed for certain conditions.
- **Documentation**
	- Updated internal comments for future error handling improvements.

### CODE REVIEW

File: packages/nocodb/src/services/datas.service.ts [hunk 1]
```diff
@@ -295,7 +295,7 @@ export class DatasService {
     });
 
     if (!row) {
-      NcError.notFound('Row not found');
+      NcError.recordNotFound(param.rowId);
```
[the remark this answers is not in the document]
Comment (mertmit) [reply packages/nocodb/src/services/datas.service.ts:298]: Formatting the message handled within the recordNotFound call, it expects identifier only

File: packages/nocodb/src/services/datas.service.ts [hunk 2]
```diff
@@ -390,7 +390,7 @@
       id: view?.fk_model_id || param.viewId,
     });
 
-    if (!model) NcError.notFound('Table not found');
+    if (!model) NcError.tableNotFound(view?.fk_model_id || param.viewId);
```
[the remark this answers is not in the document]
Comment (mertmit) [reply packages/nocodb/src/services/datas.service.ts:393]: Formatting the message handled within the recordNotFound call, it expects identifier only

File: packages/nocodb/src/services/datas.service.ts [hunk 3]
```diff
@@ -703,7 +703,7 @@
       const model = await Model.getByIdOrName({
         id: param.viewId,
       });
-      if (!model) NcError.notFound('Table not found');
+      if (!model) NcError.tableNotFound(param.viewId);
```
[the remark this answers is not in the document]
Comment (mertmit) [reply packages/nocodb/src/services/datas.service.ts:706]: Formatting the message handled within the recordNotFound call, it expects identifier only

File: packages/nocodb/src/services/datas.service.ts [hunk 4]
```diff
@@ -875,7 +875,7 @@
         titleOrId: req.params.viewName,
         fk_model_id: model.id,
       }));
-    if (!model) NcError.notFound('Table not found');
+    if (!model) NcError.tableNotFound(req.params.tableName);
```
[the remark this answers is not in the document]
Comment (mertmit) [reply packages/nocodb/src/services/datas.service.ts:878]: Formatting the message handled within the recordNotFound call, it expects identifier only

File: packages/nocodb/src/services/datas.service.ts [hunk 5]
```diff
@@ -982,8 +982,7 @@
         c.column_name === columnNameOrId,
     );
 
-    if (!column)
-      NcError.notFound(`Column with id/name '${columnNameOrId}' is not found`);
+    if (!column) NcError.fieldNotFound(columnNameOrId);
```
[the remark this answers is not in the document]
Comment (mertmit) [reply packages/nocodb/src/services/datas.service.ts:985]: Formatting the message handled within the recordNotFound call, it expects identifier only

File: packages/nocodb/src/services/columns.service.ts [hunk 6]
```diff
@@ -305,9 +305,7 @@ export class ColumnsService {
 
         await this.updateRollupOrLookup(colBody, column);
       } else {
-        NcError.notImplemented(
-          `Updating ${colBody.uidt} => ${colBody.uidt} is not implemented`,
-        );
+        NcError.notImplemented(`Updating ${colBody.uidt} => ${colBody.uidt}`);
```
[the remark this answers is not in the document]
Comment (mertmit) [reply packages/nocodb/src/services/columns.service.ts:308]: Formatting the message handled within the notImplemented call, it expects feature name only

File: packages/nocodb/src/services/columns.service.ts [hunk 7]
```diff
@@ -320,9 +318,7 @@
         UITypes.ForeignKey,
       ].includes(colBody.uidt)
     ) {
-      NcError.notImplemented(
-        `Updating ${colBody.uidt} => ${colBody.uidt} is not implemented`,
-      );
+      NcError.notImplemented(`Updating ${colBody.uidt} => ${colBody.uidt}`);
```
[the remark this answers is not in the document]
Comment (mertmit) [reply packages/nocodb/src/services/columns.service.ts:321]: Formatting the message handled within the notImplemented call, it expects feature name only

File: packages/nocodb/src/services/columns.service.ts [hunk 8]
```diff
@@ -1316,9 +1312,7 @@
           ...colBody,
         });
       } else {
-        NcError.notImplemented(
-          `Updating ${column.uidt} => ${colBody.uidt} is not supported at the moment`,
-        );
+        NcError.notImplemented(`Updating ${column.uidt} => ${colBody.uidt}`);
```
[the remark this answers is not in the document]
Comment (mertmit) [reply packages/nocodb/src/services/columns.service.ts:1315]: Formatting the message handled within the notImplemented call, it expects feature name only

File: packages/nocodb/src/services/columns.service.ts [hunk 9]
```diff
@@ -1411,9 +1405,7 @@
...
```
[the remark this answers is not in the document]
Comment (mertmit) [reply packages/nocodb/src/services/columns.service.ts:1408]: Formatting the message handled within the notImplemented call, it expects feature name only

File: packages/nocodb/src/services/columns.service.ts [hunk 10]
```diff
@@ -2271,7 +2263,7 @@
         });
         break;
       case UITypes.ForeignKey: {
-        NcError.notImplemented();
+        NcError.notImplemented(`Support for ${column.uidt}`);
```
[the remark this answers is not in the document]
Comment (mertmit) [reply packages/nocodb/src/services/columns.service.ts:2266]: Formatting the message handled within the notImplemented call, it expects feature name only

File: packages/nocodb/src/services/calendar-datas.service.ts [hunk 11]
```diff
@@ -29,7 +29,7 @@ export class CalendarDatasService {
 
     const view = await View.get(viewId);
 
-    if (!view) NcError.notFound('View not found');
+    if (!view) NcError.viewNotFound(viewId);
```
[the remark this answers is not in the document]
Comment (mertmit) [reply packages/nocodb/src/services/calendar-datas.service.ts:32]: Error handling is in global scope

File: packages/nocodb/src/services/calendar-datas.service.ts [hunk 12]
```diff
@@ -131,7 +131,7 @@
...
```
[the remark this answers is not in the document]
Comment (mertmit) [reply packages/nocodb/src/services/calendar-datas.service.ts:134]: Error handling is in global scope

File: packages/nocodb/src/modules/datas/helpers.ts [hunk 13]
```diff
@@ -37,15 +37,15 @@ export async function getViewAndModelByAliasOrId(param: {
     aliasOrId: param.tableName,
   });
 
-  if (!model) NcError.notFound('Table not found');
+  if (!model) NcError.tableNotFound(param.tableName);
```
[the remark this answers is not in the document]
Comment (mertmit) [reply packages/nocodb/src/modules/datas/helpers.ts:40]: Error handling is in global scope

File: packages/nocodb/src/modules/datas/helpers.ts [hunk 14]
```diff
@@ -37,15 +37,15 @@
...
 
   const view =
     param.viewName &&
     (await View.getByTitleOrId({
       titleOrId: param.viewName,
       fk_model_id: model.id,
     }));
-  if (param.viewName && !view) NcError.notFound('View not found');
+  if (param.viewName && !view) NcError.viewNotFound(param.viewName);
```
[the remark this answers is not in the document]
Comment (mertmit) [reply packages/nocodb/src/modules/datas/helpers.ts:48]: Error handling is in global scope

File: packages/nocodb/src/helpers/PagedResponse.ts [hunk 15]
```diff
@@ -42,7 +42,7 @@ export class PagedResponseImpl<T> {
     if (additionalProps) Object.assign(this, additionalProps);
 
     if (offset && offset >= +count) {
-      NcError.badRequest('Offset is beyond the total number of records');
+      NcError.invalidOffsetValue(`${offset}`);
```
[the remark this answers is not in the document]
Comment (mertmit) [reply packages/nocodb/src/helpers/PagedResponse.ts]: Error handling is in global scope

File: packages/nocodb/src/services/api-docs/api-docs.service.ts [hunk 16]
```diff
@@ -9,7 +9,7 @@ export class ApiDocsService {
   async swaggerJson(param: { baseId: string; siteUrl: string }) {
     const base = await Base.get(param.baseId);
 
-    if (!base) NcError.notFound();
+    if (!base) NcError.baseNotFound(param.baseId);
```
[the remark this answers is not in the document]
Comment (mertmit) [reply packages/nocodb/src/services/api-docs/api-docs.service.ts:12]: Error handling is in global scope

**Review by dstala — APPROVED** [review]

**Pull request merged**